extensions/yaml-configuration/runtime/src/main/java/io/quarkus/yaml/configuration/runtime/YamlConfigSource.java

```diff
@@ -0,0 +1,125 @@
+package io.quarkus.yaml.configuration.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.yaml.snakeyaml.Yaml;
+
+public abstract class YamlConfigSource implements ConfigSource, Serializable {
+
+    private static final long serialVersionUID = -875865410658150582L;
+
+    private static final String CONFIG_ORDINAL_KEY = "config_ordinal";
+    private static final String CONFIG_ORDINAL_DEFAULT_VALUE = "100";
+    private static final String PROPERTY_KEY_SEPARATOR = ".";
+    private static final String COMMA = ",";
+    private static final String ESCAPED_COMMA = "\\\\,";
+
+    private final Map<String, String> properties;
+    private final String source;
+    private final int ordinal;
+
+    protected YamlConfigSource(String source, int ordinal) {
+        this.properties = buildConfigPropertiesFromYaml();
+        this.source = source;
+        if (properties.containsKey(CONFIG_ORDINAL_KEY)) {
```

Comment:
It seems to be used in the `toString` only? Maybe  pass the URL or a Path and then you don't need the `openIYamlFilestream` method

Reply:
Also you won't need two implementation classes with a URL or a Path in the constructor :)